openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -795,6 +795,22 @@ def get_parameter_io_handler(self, io_format):
 
         return io_handler
 
+    def delete_parameter_handler(self, handler):
+        """
+        Delete a parameter handler specified by tag name, class, or instance.
+
+        Parameters
+        ----------
+        handler: str, openforcefield.typing.engines.smirnoff.ParameterHandler-derived type or object
+            The handler to delete.
+        """
+        if isinstance(handler, ParameterHandler):
+            tagname = handler.TAGNAME
+        elif isinstance(handler, type):
+            tagname = handler._TAGNAME
+        else:
```

Comment:
You might consider adding a fallback with a more informative exception 
```suggestion
        elif isinstance(tagname, str):
            tagname = handler
        else:
            raise InvalidHandlerKey(handler) 
```
however you could also just use a built-in exception like `NotImplementedError` for now if you don't have time!